features/netsocket/SocketAddress.cpp

```diff
@@ -290,6 +290,8 @@ bool operator==(const SocketAddress &a, const SocketAddress &b)
     } else if (a._addr.version == NSAPI_IPv6) {
         return memcmp(a._addr.bytes, b._addr.bytes, NSAPI_IPv6_BYTES) == 0;
     }
+
+    MBED_UNREACHABLE;
```

Comment:
Well, a valid program should never actually reach this state after the previous conditions. We should probably make MBED_UNREACHABLE assert in debug builds.